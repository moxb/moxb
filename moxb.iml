<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/packages/antd/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/meteor/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/moxb/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/react-html/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/examples/src" isTestSource="false" />
      <excludeFolder url="file://$MODULE_DIR$/packages/moxb/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/meteor/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/react-html/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/antd/dist" />
      <excludeFolder url="file://$MODULE_DIR$/admin/node-installation" />
      <excludeFolder url="file://$MODULE_DIR$/admin/bin-tools" />
      <excludeFolder url="file://$MODULE_DIR$/examples/dist" />
      <excludeFolder url="file://$MODULE_DIR$/examples/.cache" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>