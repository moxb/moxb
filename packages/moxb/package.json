{
  "name": "@moxb/moxb",
  "version": "0.2.0-beta.59",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "dist",
  "typings": "dist/",
  "files": [
    "dist/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ],
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest"
    },
    "testEnvironment": "node",
    "testMatch": [
      "**/src/**/__tests__/*.(ts|tsx)"
    ],
    "modulePaths": [
      "<rootDir>/peer-dependencies/node_modules"
    ]
  },
  "peerDependencies": {
    "mobx": "^5.15.6",
    "tslib": "^2.0.1"
  },
  "devDependencies": {
    "@types/atob": "2.1.2",
    "@types/jest": "26.0.13",
    "@types/lodash.flatten": "4.4.6",
    "@types/urijs": "1.19.11",
    "mobx": "5.13.1",
    "tslib": "1.10.0"
  },
  "dependencies": {
    "btoa-lite": "1.0.0",
    "history": "5.0.0",
    "lodash.flatten": "4.4.0",
    "moment": "2.29.1",
    "urijs": "1.19.2"
  }
}
