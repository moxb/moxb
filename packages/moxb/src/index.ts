export * from './impl';
export * from './types';
export * from './routing';
