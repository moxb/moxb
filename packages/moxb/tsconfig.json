{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["dom","es6","es2015","es2017", "dom.iterable","scripthost"],
        "inlineSourceMap": true,
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "./dist",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "allowUnreachableCode": false,
        "baseUrl": "./src",
        "paths": {
            "*": [
                "*",
                "../../../node_modules/*"
            ]
        }
    },
    "exclude": [
        "./node_modules",
        "./dist"
    ]
}
