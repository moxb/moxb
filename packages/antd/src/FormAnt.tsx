import { Form as MoxForm, t } from '@moxb/moxb';
import { Alert, Form } from 'antd';
import { FormProps } from 'antd/lib/form/Form';
import { observer } from 'mobx-react';
import * as React from 'react';
import { parseProps } from './BindAnt';

export interface BindFormAntProps extends React.HTMLProps<HTMLFormElement> {
    operation: MoxForm;
    hideErrors?: boolean;
}

@observer
export class FormAnt extends React.Component<FormProps & BindFormAntProps> {
    render() {
        const { operation, children, invisible, hideErrors, ...props } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }

        return (
            <Form data-testid={operation.id} onFinish={operation.onSubmitForm} {...(props as any)}>
                {children}
                {!hideErrors && operation.hasErrors && (
                    <Alert
                        message={t('FormAnt.errors.header', 'Errors')}
                        description={operation.allErrors.join(', ')}
                        type="error"
                        closable
                        onClose={operation.clearErrors}
                    />
                )}
            </Form>
        );
    }
}
