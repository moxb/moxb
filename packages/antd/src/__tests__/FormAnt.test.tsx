import { FormImpl, TextImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { FormAnt } from '../FormAnt';
import { shallowMoxbToJson } from './enzymeHelper';

describe('FormAnt', function () {
    const onSaveUserText = jest.fn();
    const onSavePasswordText = jest.fn();
    const onSubmitMock = jest.fn();
    const formUserText = new TextImpl({
        id: 'id.formUserText',
        label: 'Username',
        onSave: onSaveUserText,
    });
    const formPasswordText = new TextImpl({
        id: 'id.formUserText',
        label: 'Username',
        onSave: onSavePasswordText,
    });

    function newFormOperation(...opt: any[]) {
        return new FormImpl(
            Object.assign(
                {
                    id: 'id.testForm',
                    values: [formUserText, formPasswordText],
                    onSubmit: onSubmitMock,
                },
                ...opt
            )
        );
    }

    it('should render a form element by default', function () {
        const operation = newFormOperation();
        expect(shallowMoxbToJson(shallow(<FormAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should throw if no values for children were defined', function () {
        expect(() => {
            new FormImpl({
                id: 'id.testForm',
                values: [],
                onSubmit: onSubmitMock,
            });
        }).toThrow();
    });

    it('should return null if invisible', function () {
        const operation = newFormOperation({ invisible: () => true });
        expect(shallow(<FormAnt operation={operation} />).type()).toBeNull();
    });

    it('should show an alert, if an error happens in a child component', function () {
        const operation = newFormOperation();
        formUserText.setError('New error');
        const wrapper = shallow(<FormAnt operation={operation} />);
        expect(wrapper.render().find('div.ant-alert').length).toBe(1);
    });

    it('should not show an alert on error with property "hideErrors" set', function () {
        const operation = newFormOperation();
        formUserText.setError('New error');
        const wrapper = shallow(<FormAnt hideErrors operation={operation} />);
        expect(wrapper.render().find('div.ant-alert').length).toBe(0);
    });

    it('should show an alert, if an error in the form happened', function () {
        const operation = newFormOperation();
        operation.setError('New error');
        const wrapper = shallow(<FormAnt operation={operation} />);
        expect(wrapper.render().find('div.ant-alert').length).toBe(1);
    });
});
