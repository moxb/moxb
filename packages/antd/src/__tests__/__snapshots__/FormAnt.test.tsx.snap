// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAnt should render a form element by default 1`] = `
<Form
  colon={true}
  data-testid="id.testForm"
  disabled={false}
  hideRequiredMark={false}
  id="id-test_form"
  layout="horizontal"
  onSubmit={[Function]}
  readOnly={false}
/>
`;
